rllib/algorithms/callbacks.py

```diff
@@ -945,10 +945,18 @@ def on_postprocess_trajectory(
 
         @override(DefaultCallbacks)
         def on_sample_end(
-            self, *, worker: "EnvRunner", samples: SampleBatch, **kwargs
```

Comment:
This was the bug: `MultiAgentEnvRunner` and `SingleAgentEnvRunner` call this without the `worker` arg (but then `env_runner` is provided), leading to a
```
TypeError: make_multi_callbacks.<locals>._MultiCallbacks.on_sample_end() missing 1 required keyword-only argument: 'worker'
```